src/http-schema/http-schema.test.ts

```diff
@@ -0,0 +1,127 @@
+import { EnhancedArticleSchema } from './http-schema';
+
+const enhancedArticleExample = {
+  id: 'testid1',
+  msid: 'testmsid1',
+  doi: 'doi1',
+  versionIdentifier: '1',
+  versionDoi: 'publisher/testid1',
+  article: {
+    doi: 'preprint/testid1',
+    title: 'test article',
+    date: '2023-01-02',
+    authors: [
+      {
+        familyNames: ['Daffy'],
+        givenNames: ['Duck'],
+        affiliations: [{ name: 'ACME Labs' }],
+        emails: ['daffy.duck@acme.org'],
+      },
+    ],
+    abstract: 'This is the test abstract',
+    licenses: [],
+    content: 'This is some test content',
+    headings: [{ id: 'head1', text: 'Heading 1' }],
+    references: [],
+  },
+  preprintDoi: 'preprint/testid1',
+  preprintUrl: 'doi.org/preprint/testid1',
+  preprintPosted: '2023-01-02',
+  sentForReview: '2023-01-03',
+  published: '2023-01-23',
+};
+
+describe('httpschema', () => {
+  it.each([
+    'foo',
+    ['one', 'two', { type: 'Strong', content: 'three' }],
+    {
+      type: 'Heading', depth: 1, content: 'heading', id: 'h1',
+    },
+    {
+      type: 'Cite',
+      content: 'I am a citation',
+      target: 'target',
+    },
+    {
+      type: 'Link',
+      content: 'I am a link',
+      target: 'target',
+    },
+    {
+      type: 'Paragraph',
+      content: 'I am a paragraph',
+    },
+    {
+      type: 'Emphasis',
+      content: 'I am emphasised',
+    },
+    {
+      type: 'Strong',
+      content: 'I am strong',
+    },
+    {
+      type: 'Superscript',
+      content: 'I am super',
+    },
+    {
+      type: 'Subscript',
+      content: 'I am a subscript',
+    },
+    {
+      type: 'Date',
+      content: '13/01/2001',
+    },
+    {
+      type: 'Figure',
+      content: 'I am a figure',
+      caption: 'I am a caption',
+      label: 'I am a label',
+      id: 'id',
+    },
+    {
+      type: 'ImageObject',
+      contentUrl: 'https://placekitten.com/500/300',
+      content: [],
+      meta: {
+        inline: false,
+      },
+    },
+    [
+      [{
+        type: 'Heading', depth: 1, content: 'heading', id: 'h1',
+      }],
+    ],
+  ])('validates %s content', (value) => {
+    const enhancedArticle = {
+      ...enhancedArticleExample,
+      article: {
+        ...enhancedArticleExample.article,
+        content: value,
+      },
+    };
+    expect(EnhancedArticleSchema.validate(enhancedArticle).error).toBeUndefined();
+  });
+
+  it.each([
+    [{id: '12345', msid: 1}, '"msid" must be a string'],
+    [{unknown: 'unknown'}, '"unknown" is not allowed'],
+    [{unknown1: 'unknown', unknown2: 'unknown'}, '"unknown1" is not allowed'],
+  ])('handles validation error', (value, error) => {
+    const invalidateEnhancedArticle = EnhancedArticleSchema.validate(value);
+
+    expect(invalidateEnhancedArticle.error).toBeDefined();
+    expect(invalidateEnhancedArticle.error?.details).toHaveLength(1);
```

Comment:
Is it expected that only the first error is detailed?

Reply:
You can pass an optional second param `{ abortEarly: false }` to `validate()` that will return the full validation chain (default behaviour of `abortEarly: true` is to exit on first validation). I pushed a commit amounting to that here.

Separately though, I would question those last two examples. They are mostly testing the validation library, rather than our particular validation rules. The first one seems fine though.

Reply:
What's even more interesting I think is that missing fields aren't reported. This feels crucial, now that I think about it. @will-byrne - something to work on today?

Reply:
@scottaubrey the `abortEarly: false` is great. I didn't want to contribute anything that would make the PR not green because this is adding value now. The intention was just to introduce the "fail validation" scenario.

We also need to ensure that we are logging the validation errors and that we preserve the data that is failing validation so that we can adequately address failed requests.